taf/testlib/linux/service_lib.py

```diff
@@ -133,71 +134,133 @@ def method(service_name):
 
 
-class SpecificServiceManager(object):
+class GenericServiceManager(object):
+    def __init__(self, run_func, command_list=None):
+        super().__init__()
+        if command_list is None:
+            command_list = COMMANDS
+        self.service_command_generator = ServiceCommandGenerator(systemd_command_generator,
+                                                                 SystemdReturnCodes,
+                                                                 command_list)
+
+        self.return_codes = SystemdReturnCodes
+        self.run_func = run_func
+
+    def __getattr__(self, name):
+        def run(service='', **kwargs):
+            return self.run_func(' '.join(command(service)), **kwargs)
+        command = getattr(self.service_command_generator, name)
+        setattr(self, name, run)
+        return run
 
-    def __init__(self, service_name, service_command_generator, run):
-        super(SpecificServiceManager, self).__init__()
-        for cmd in service_command_generator.commands:
-            setattr(self, cmd,
-                    self.generate_run_function(run, getattr(service_command_generator, cmd), service_name))
+    def _get_running_status(self, service=''):
+        return self.status(service=service, expected_rcs={self.return_codes.RUNNING,
+                                                          self.return_codes.STOPPED})
 
-    @staticmethod
-    def generate_run_function(run_func, command, service_name):
-        def run(**kwargs):
-            return run_func(" ".join(command(service_name)), **kwargs)
-        return run
+    def is_running(self, service=''):
+        rv = self._get_running_status(service)
+        return rv.rc == self.return_codes.RUNNING
 
+    def is_stopped(self, service=''):
+        rv = self._get_running_status(service)
+        return rv.rc == self.return_codes.STOPPED
 
-class GenericServiceManager(object):
 
-    def __init__(self, service_command_generator, run):
-        super(GenericServiceManager, self).__init__()
-        for cmd in service_command_generator.commands:
-            setattr(self, cmd,
-                    self.generate_run_function(run, getattr(service_command_generator, cmd)))
+class SpecificServiceManager(GenericServiceManager):
+    def __init__(self, service_name, run_func):
+        command_list = [c for c in COMMANDS if c != "list"]
+        super().__init__(run_func, command_list)
+        self.service_name = service_name
 
-    @staticmethod
-    def generate_run_function(run_func, command):
-        def run(service="", **kwargs):
-            return run_func(" ".join(command(service)), **kwargs)
+    def __getattr__(self, name):
+        def run(**kwargs):
+            kwargs.pop('service', None)  # remove any value associated with the service key
+            return self.run_func(command, **kwargs)
+        command = getattr(self.service_command_generator, name)
+        command = ' '.join(command(self.service_name))
```

Comment:
is there a reason to do this join when the command execution function should do it for us?